gnd/src/main/java/com/google/android/gnd/persistence/local/room/RecordMutationEntity.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.local.room;
+
+import static androidx.room.ForeignKey.CASCADE;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.Index;
+import androidx.room.PrimaryKey;
+import com.google.android.gnd.persistence.shared.RecordMutation;
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.AutoValue.CopyAnnotations;
+import org.json.JSONObject;
+
+/**
+ * Representation of a {@link com.google.android.gnd.persistence.shared.RecordMutation} in local
+ * data store.
+ */
+@AutoValue
+@Entity(
+    tableName = "record_mutation",
+    foreignKeys =
+        @ForeignKey(
+            entity = RecordEntity.class,
+            parentColumns = "id",
+            childColumns = "record_id",
+            onDelete = CASCADE),
+    indices = {@Index("record_id")})
+public abstract class RecordMutationEntity {
+  @CopyAnnotations
+  @PrimaryKey(autoGenerate = true)
+  @ColumnInfo(name = "id")
+  @Nullable
+  public abstract Long getId();
+
+  @CopyAnnotations
+  @NonNull
+  @ColumnInfo(name = "record_id")
+  public abstract String getRecordId();
+
+  @CopyAnnotations
+  @NonNull
+  @ColumnInfo(name = "type")
+  public abstract MutationEntityType getType();
+
+  /**
+   * For edits of type {@link MutationEntityType#CREATE} and {@link MutationEntityType#UPDATE}, a
+   * JSON object with the new value of modified attributes after this mutation. For all other
+   * mutation types this will be null.
+   *
+   * <p>Null values in the responses indicates a response was removed/cleared.
```

Comment:
Ah, good catch! One way around this would be to return an empty `JSONObject` for `DELETE` mutations, but this is semantically inconsistent, since following the same rules it would be interpreted as "nothing changed".

Representing removed responses separately or using a special placeholder is also an option, but trades one kind of complexity for another. 

For now I've tried to clarify the Javadoc; please resolve if you think it works.

Reply:
I think the modified doc is great!

We can rethink the technical approach if it proves problematic in the future (in other words, if people don't read the doc string).